x-pack/plugins/reporting/server/export_types/csv_searchsource/generate_csv/generate_csv.ts

```diff
@@ -0,0 +1,400 @@
+  private csvContainsFormulas = false;
+  private maxSizeReached = false;
+  private csvRowCount = 0;
+
+  constructor(
+    private job: JobParamsCSV,
+    private config: ReportingConfig,
+    private clients: Clients,
+    private dependencies: Dependencies,
+    private cancellationToken: CancellationToken,
+    private logger: LevelLogger
+  ) {}
+
+  private async scan(
+    index: IndexPattern,
+    searchSource: ISearchSource,
+    scrollSettings: CsvExportSettings['scroll']
+  ) {
+    const searchBody = await searchSource.getSearchRequestBody();
+    this.logger.debug(`executing search request`);
+    const searchParams = {
+      params: {
+        body: searchBody,
+        index: index.title,
+        scroll: scrollSettings.duration,
+        size: scrollSettings.size,
+      },
+    };
+    const results = (
+      await this.clients.data.search(searchParams, { strategy: ES_SEARCH_STRATEGY }).toPromise()
+    ).rawResponse;
+
+    return results;
+  }
+
+  private async scroll(scrollId: string, scrollSettings: CsvExportSettings['scroll']) {
+    this.logger.debug(`executing scroll request`);
+    const results = (
+      await this.clients.es.asCurrentUser.scroll({
+        scroll: scrollSettings.duration,
+        scroll_id: scrollId,
+      })
+    ).body as SearchResponse<unknown>;
+    return results;
+  }
+
+  /*
+   * Load field formats for each field in the list
+   */
+  private getFormatters(table: Datatable) {
+    if (this._formatters) {
+      return this._formatters;
+    }
+
+    // initialize field formats
+    const formatters: Record<string, FieldFormat> = {};
+    table.columns.forEach((c) => {
+      const fieldFormat = this.dependencies.fieldFormatsRegistry.deserialize(c.meta.params);
+      formatters[c.id] = fieldFormat;
+    });
+
+    this._formatters = formatters;
+    return this._formatters;
+  }
+
+  private escapeValues(settings: CsvExportSettings) {
+    return (value: string) => {
+      if (settings.checkForFormulas && cellHasFormulas(value)) {
+        this.csvContainsFormulas = true; // set warning if cell value has a formula
+      }
+      return settings.escapeValue(value);
+    };
+  }
+
+  // use fields/fieldsFromSource from the searchSource to get the ordering of columns
+  // otherwise use the table columns as they are
+  private getFields(searchSource: ISearchSource, table: Datatable): string[] {
+    const fieldValues: Record<string, string | boolean | SearchFieldValue[] | undefined> = {
+      fields: searchSource.getField('fields'),
+      fieldsFromSource: searchSource.getField('fieldsFromSource'),
+    };
+    const fieldSource = fieldValues.fieldsFromSource ? 'fieldsFromSource' : 'fields';
+    this.logger.debug(`Getting search source fields from: '${fieldSource}'`);
+
+    const fields = fieldValues[fieldSource];
+    // Check if field name values are string[] and if the fields are user-defined
+    if (isPlainStringArray(fields)) {
```

Comment:
why do we try to first parse fields from searchsource instead of always just using the ones in the table ?

Reply:
if its tabify dropping fields that don't exist in response (but do exist in searchsource) and discover is actually showing those fields, then we should probably update tabify.